plugins/org.eclipse.sirius.ui/src/org/eclipse/sirius/ui/business/internal/session/EditingSession.java

```diff
@@ -532,7 +533,7 @@ public boolean needToBeSavedOnClose(IEditorPart editor) {
         }
 
         public boolean closeAllDetected() {
-            return closingEditors.size() == editors.size();
+            return !closingEditors.isEmpty() && closingEditors.size() == editors.size();
```

Comment:
Not sure what this change does/why it is needed.

Reply:
When sizes are equals but equals to 0, i.e. no closing editors and no opened editors: it is not a closeAll, and we must not "promptToSaveOnClose".

I admit that this was my first patch, but then I have added the second commit with "restoreToSavePointListener!=null" guard which  is more generic and hide this empty closingEditors case.